### PR TITLE
Add class to widgets with current number of columns

### PR DESCRIPTION
**Pull request by CodyWMitchell**
[RHCLOUD-31434](https://issues.redhat.com/browse/RHCLOUD-31434)

Adds the class `widget-columns-#`  to widgets with the current # of columns being occupied by the widget in order to style the widgets based on size. This class is updated whenever the layout is changed.

### CODE REVIEW

**Review by epwinchell — APPROVED** [review]

**Comment by epwinchell** [comment]
@CodyWMitchell It seems to be working like a charm. Thank you!

File: src/Components/DnDLayout/GridLayout.tsx [hunk 1]
```diff
@@ -332,6 +345,7 @@ const GridLayout = ({ isLayoutLocked = false }: { isLayoutLocked?: boolean }) =>
                 boxShadow: activeItem === rest.i ? '0 0 2px 2px #2684FF' : 'none',
                 ...(activeItem === rest.i ? { outline: 'none' } : {}),
               }}
+              className={widgetClasses[rest.i] || ''}
```
Comment (Hyperkid123) [inline src/Components/DnDLayout/GridLayout.tsx:348]: No need to fallback to an empty string.

File: src/Components/DnDLayout/GridLayout.tsx [hunk 2]
```diff
@@ -119,6 +124,14 @@ const GridLayout = ({ isLayoutLocked = false }: { isLayoutLocked?: boolean }) =>
     [isLayoutLocked, layout]
   );
 
+  useEffect(() => {
+    const updatedWidgetClasses = layout.reduce<WidgetClasses>((acc, curr) => {
+      acc[curr.i] = `widget-columns-${curr.w}`;
+      return acc;
+    }, {});
+    setWidgetClasses(updatedWidgetClasses);
+  }, [layout]);
```
Comment (florkbr) [inline src/Components/DnDLayout/GridLayout.tsx:133]: @CodyWMitchell I'd be curious how often this gets fired as the user interacts with the page  - and if we have a way of reducing the amount of times the effect is called. We really only care when the number of columns changes right? I wonder if we should listen for when the `layoutVariant` changes instead, then iterate over the layout to update the classes 🤔

**Review by Hyperkid123 — COMMENTED** [review]
@CodyWMitchell 

Do we need the classes at all? Can't we just pass the number of columns down to the grid tile parent div and set it there? As the class name prefix is not changing you should have the width available in the map.

Isn't this line of code achieve the same thing? (Maybe I am missing something)
```diff
diff --git a/src/Components/DnDLayout/GridLayout.tsx b/src/Components/DnDLayout/GridLayout.tsx
index 46ade80..8a206ee 100644
--- a/src/Components/DnDLayout/GridLayout.tsx
+++ b/src/Components/DnDLayout/GridLayout.tsx
@@ -333,6 +333,7 @@ const GridLayout = ({ isLayoutLocked = false }: { isLayoutLocked?: boolean }) =>
                 boxShadow: activeItem === rest.i ? '0 0 2px 2px #2684FF' : 'none',
                 ...(activeItem === rest.i ? { outline: 'none' } : {}),
               }}
+              className={`widget-columns-${rest.w}`}
             >
               <GridTile
                 isDragging={isDragging}

```

**Comment by CodyWMitchell** [comment]
> @CodyWMitchell
> 
> Do we need the classes at all? Can't we just pass the number of columns down to the grid tile parent div and set it there? As the class name prefix is not changing you should have the width available in the map.
> 
> Isn't this line of code achieve the same thing? (Maybe I am missing something)
> 
> ```diff
> diff --git a/src/Components/DnDLayout/GridLayout.tsx b/src/Components/DnDLayout/GridLayout.tsx
> index 46ade80..8a206ee 100644
> --- a/src/Components/DnDLayout/GridLayout.tsx
> +++ b/src/Components/DnDLayout/GridLayout.tsx
> @@ -333,6 +333,7 @@ const GridLayout = ({ isLayoutLocked = false }: { isLayoutLocked?: boolean }) =>
>                  boxShadow: activeItem === rest.i ? '0 0 2px 2px #2684FF' : 'none',
>                  ...(activeItem === rest.i ? { outline: 'none' } : {}),
>                }}
> +              className={`widget-columns-${rest.w}`}
>              >
>                <GridTile
>                  isDragging={isDragging}
> ```

On my end that adds the correct class initially, but it doesn't update as the tile is resized.

**Comment by Hyperkid123** [comment]
@CodyWMitchell  Hmm, that sounds strange. The layout is persisted on a tile change which should trigger a re-render. Would you have some time tomorrow to talk about it a bit more?

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**